src/MIDebugEngine/Natvis.Impl/Natvis.cs

```diff
@@ -191,16 +191,44 @@ public void Initialize(string fileName)
             {
                 // failed to find the VS Service
             }
-            string natvisPath = null;
-            string directory = _process.Engine.GetMetric("GlobalVisualizersDirectory") as string;
-            if (!string.IsNullOrEmpty(directory) && !string.IsNullOrEmpty(fileName))
+
+            if (string.IsNullOrEmpty(Path.GetDirectoryName(fileName)))
             {
-                natvisPath = Path.Combine(directory, fileName);
-            }
+                string globalVisualizersDirectory = _process.Engine.GetMetric("GlobalVisualizersDirectory") as string;
+                string globalNatVisPath = null;
+                if (!string.IsNullOrEmpty(globalVisualizersDirectory) && !string.IsNullOrEmpty(fileName))
+                {
+                    globalNatVisPath = Path.Combine(globalVisualizersDirectory, fileName);
+                }
 
-            if (!string.IsNullOrEmpty(natvisPath))
+                // For local launch, try and load natvis next to the target exe if it exists.
+                if (_process.LaunchOptions is LocalLaunchOptions)
+                {
+                    string natvisPath = Path.Combine(Path.GetDirectoryName((_process.LaunchOptions as LocalLaunchOptions).ExePath), fileName);
+
+                    if (File.Exists(natvisPath))
+                    {
+                        LoadFile(natvisPath);
+                        return;
+                    }
+                    else if (!File.Exists(globalNatVisPath))
```

Comment:
I don't have the full context on how natvis files are loaded, but it seems a bit strange that:

```
If the launch options are local {
- We try the local one if it exists and return
- Check if the global one exists and show error if it doesn't
}

- Load the global one if it exists

```
It would be more clear:

```
fileToLoad = globalNatVisPath
If the launch options are local {
- Set fileToLoad to natvisPath if natvisPath exists (natvisPath probably should be renamed to localNatVisPath)
}

- Load fileToLoad if it exists, show error if it doesn't

```